build.rs

```diff
@@ -0,0 +1,17 @@
+fn main() {
+    let git_describe = std::process::Command::new("git")
+        .args(&["describe", "--tags", "--broken"])
+        .output()
+        .ok()
+        .and_then(|proc| String::from_utf8(proc.stdout).ok());
+
+    let mut version_info = format!("v{}-unknown", env!("CARGO_PKG_VERSION"));
```

Comment:
I did it this way because otherwise, there'd be two `println!`s (out of three total) with the same contents: one in the case that `description` is empty (e.g. the command failed -- it prints to stderr and has an empty stdout) and one in the case that `git` was not found (or any other reason that would cause the process or its stdout to be `None`).

Reply:
You could probably do this instead:

```rust
let version_info = match git_describe {
    Some(description) if !description.is_empty() => description,
    _ => format!("v{}-unknown", env!("CARGO_PKG_VERSION"));,
}
```

Reply:
Hah, I keep forgetting you can do that stuff inside `match` expressions. Thanks for reminding me.

Reply:
FYI, if there's only one condition to match, you can also use the `matches!` macro as an if condition!